src/hist/plot.py

```diff
@@ -212,16 +220,230 @@ def plot2d_full(
+        if len(available[0]) == 0:
+            raise RuntimeWarning(
+                "All values are zero! Cannot compute meaningful error bars",
+            )
+            return np.vstack([values, values])
+        nearest = np.sum(
+            [np.square(np.subtract.outer(d, d0)) for d, d0 in zip(available, missing)]
+        ).argmin(axis=0)
+        argnearest = tuple(dim[nearest] for dim in available)
+        scale[missing] = scale[argnearest]
+    counts = values / scale
+    interval_min = scale * stats.chi2.ppf((1 - coverage) / 2, 2 * counts) / 2.0
+    interval_max = scale * stats.chi2.ppf((1 + coverage) / 2, 2 * (counts + 1)) / 2.0
+    interval = np.array([interval_min, interval_max])
+    interval[interval == np.nan] = 0.0  # chi2.ppf produces nan for counts=0
+    return interval
+
+
+def clopper_pearson_interval(
+    num: np.ndarray, denom: np.ndarray, coverage: "Optional[float]" = None
+) -> np.ndarray:
+    """Compute Clopper-Pearson coverage interval for a binomial distribution
+    Parameters
+    ----------
+        num : numpy.ndarray
+            Numerator, or number of successes, vectorized
+        denom : numpy.ndarray
+            Denominator or number of trials, vectorized
+        coverage : float, optional
+            Central coverage interval, defaults to 68%
+    c.f. http://en.wikipedia.org/wiki/Binomial_proportion_confidence_interval
+    """
+    if coverage is None:
+        coverage = stats.norm.cdf(1) - stats.norm.cdf(-1)
+    # Numerator is subset of denominator
+    if np.any(num > denom):
+        raise ValueError(
+            "Found numerator larger than denominator while calculating binomial uncertainty"
+        )
+    interval_min = stats.beta.ppf((1 - coverage) / 2, num, denom - num + 1)
+    interval_max = stats.beta.ppf((1 + coverage) / 2, num + 1, denom - num)
+    interval = np.array([interval_min, interval_max])
+    interval[:, num == 0.0] = 0.0
+    interval[1, num == denom] = 1.0
+    return interval
+
+
+# TODO: Why is return type "Any"?
+def ratio_uncertainty(
+    num: np.ndarray, denom: np.ndarray, uncert_type: str = "poisson"
+) -> Any:
+    with np.errstate(divide="ignore"):
+        ratio = num / denom
+    if uncert_type == "poisson":
+        ratio_uncert = np.abs(poisson_interval(ratio, num / np.square(denom)) - ratio)
+        # TODO: Why does SciPy not work here?
+        # coverage = stats.norm.cdf(1) - stats.norm.cdf(-1)
+        # alpha = 1 - coverage
+        # ratio_uncert = stats.poisson.interval(alpha, ratio)
+    elif uncert_type == "poisson-ratio":
+        # poisson ratio n/m is equivalent to binomial n/(n+m)
+        ratio_uncert = np.abs(clopper_pearson_interval(num, num + denom) - ratio)
+    return ratio_uncert
+
+
+# TODO: FIX TYPING
+def _fit_callable_to_hist(
+    model: Callable[[np.ndarray], np.ndarray],
+    _hist: hist.BaseHist,
+    likelihood: bool = False,
+) -> "Tuple[np.ndarray, np.ndarray, np.ndarray, Tuple[Tuple[np.float64, ...], np.ndarray]]":
```

Comment:
I have confused myself on typing again as I don't understand why my tuple with a tuple of floats is wrong.

```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

src/hist/plot.py:351: error: Incompatible return value type (got "Tuple[ndarray, Any, Any, Tuple[Tuple[float, ...], Union[Union[int, float, complex, str, bytes, generic], Sequence[Union[int, float, complex, str, bytes, generic]], Sequence[Sequence[Any]], _SupportsArray]]]", expected "Tuple[ndarray, ndarray, ndarray, Tuple[Tuple[floating[_64Bit], ...], ndarray]]")
Found 1 error in 1 file (checked 17 source files)
```

Reply:
```suggestion
) -> Tuple[np.ndarray, Any, Any, Tuple[Tuple[float, ...], "ArrayLike"]]:
```

Should be just fine. But better yet, fix the Any's and ArrayLike, they are too broad for a good return.

Reply:
Rebase or merge master in, and this should be better. I've swapped ArrayLike for np.ndarray, it always was an ndarray anyway.